### PR TITLE
[fixes] set parent item group by checking lft = 1 and set default sup…

### PR DESCRIPTION
**Pull request by saurabh6790**
…plier to item while syncing

### CODE REVIEW

File: erpnext_shopify/patches/V2_0/set_default_supplier.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import frappe
+from erpnext_shopify.shopify_requests import get_shopify_items
+from erpnext_shopify.sync_products import get_supplier
+
+def execute():
+	for shopify_item in get_shopify_items(ignore_filter_conditions=True):
+		name = frappe.db.get_value("Item", {"shopify_product_id": shopify_item.get("id")}, "name")
+		if name:
+			frappe.db.set_value("Item", name, "default_supplier", get_supplier(shopify_item), update_modified=False)
+	frappe.db.commit()
```
Comment (anandpdoshi) [inline erpnext_shopify/patches/V2_0/set_default_supplier.py]: Did you intend to put commit inside the loop?

File: erpnext_shopify/sync_products.py [hunk 2]
```diff
@@ -164,18 +165,20 @@ def get_attribute_value(variant_attr_val, attribute):
...
 				"item_group_name": product_type,
-				"parent_item_group": _("All Item Groups"),
+				"parent_item_group": frappe.db.get_value("Item Group", {"lft": 1}, "name"),
 				"is_group": "No"
 			}).insert().name
 		else:
 			return product_type
 	else:
-		return _("All Item Groups")
+		return frappe.db.get_value("Item Group", {"lft": 1}, "name")
```
Comment (anandpdoshi) [inline erpnext_shopify/sync_products.py]: Use: `frappe.utils.nestedset.get_root_of("Item Group")`

File: erpnext_shopify/sync_products.py [hunk 3]
```diff
@@ -201,6 +204,19 @@ def get_item_image(shopify_item):
...
 	return None
 
+def get_supplier(shopify_item):
+	if shopify_item.get("vendor"):
+		if not frappe.db.get_value("Supplier", {"supplier_name": shopify_item.get("vendor")}, "name"):
+			return frappe.get_doc({
+				"doctype": "Supplier",
+				"supplier_name": shopify_item.get("vendor"),
+				"supplier_type": frappe.db.sql("select name from `tabSupplier Type` limit 1")[0][0]
+			}).insert().name
```
Comment (anandpdoshi) [inline erpnext_shopify/sync_products.py]: Too much chaining

Use 

```
supplier = frappe.get_doc({...}).insert()
return supplier.name
```

File: erpnext_shopify/sync_products.py [hunk 4]
```diff
@@ -201,6 +205,20 @@ def get_item_image(shopify_item):
 		return shopify_item.get("image").get("src")
 	return None
 
+def get_supplier(shopify_item):
+	if shopify_item.get("vendor"):
+		if not frappe.db.get_value("Supplier", {"supplier_name": shopify_item.get("vendor")}, "name"):
```
Comment (nabinhait) [inline erpnext_shopify/sync_products.py]: Maintain separate `shopify_supplier_id` field and set supplier type as root

File: erpnext_shopify/patches/V2_0/set_default_supplier.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
+# License: GNU General Public License v3. See license.txt
+import frappe
+from erpnext_shopify.shopify_requests import get_shopify_items
+from erpnext_shopify.sync_products import get_supplier
+
+def execute():
+	for index, shopify_item in enumerate(get_shopify_items(ignore_filter_conditions=True)):
+		name = frappe.db.get_value("Item", {"shopify_product_id": shopify_item.get("id")}, "name")
+		if name:
```
Comment (nabinhait) [inline erpnext_shopify/patches/V2_0/set_default_supplier.py]: if get_supplier(shopify_item)

File: erpnext_shopify/sync_products.py [hunk 6]
```diff
@@ -164,18 +165,21 @@ def get_attribute_value(variant_attr_val, attribute):
...
+	# get parent item group by checking lft = 1
+	
 	if product_type:
 		if not frappe.db.get_value("Item Group", product_type, "name"):
-			return frappe.get_doc({
+			item_group = frappe.get_doc({
 				"doctype": "Item Group",
 				"item_group_name": product_type,
-				"parent_item_group": _("All Item Groups"),
+				"parent_item_group": frappe.utils.nestedset.get_root_of("Item Group"),
```
Comment (nabinhait) [inline erpnext_shopify/sync_products.py]: set parent_item_group in a variable

**Comment by saurabh6790** [comment]
https://github.com/frappe/erpnext_shopify/issues/48
https://github.com/frappe/erpnext_shopify/issues/40

**Pull request merged**